gtn/functions/compose.cpp

```diff
@@ -168,7 +197,16 @@ void addEpsilonReachableNodes(
         ngraph);
 
     if (isReachable) {
-      gradInfo.emplace_back(-1, i);
+      // The edge with this corresponding gradInfo index in the composed graph
+      // corresponds to index i in the first/second graph. Wlog, if advancing
+      // along an epsilon edge in the first graph, shouldn't have the gradient
+      // corresponding to the resulting edge in the composed graph applied to it
+      // at backwards time.
```

Comment:
Nit: this comment is not quite right. The arc that we are advancing on is the one which needs a gradient. So it should say:

"Only apply a gradient for the epsilon edge in the graph in which we are advancing."

Reply:
Yep, I flipped it. Will fix it at some point. Thanks for catching.